Sources/BraveWallet/Crypto/Accounts/AccountSelectionRootView.swift

```diff
@@ -0,0 +1,135 @@
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveCore
+import struct Shared.Strings
+import BraveUI
+
+struct AccountSelectionRootView: View {
+  
+  let navigationTitle: String
+  let allAccounts: [BraveWallet.AccountInfo]
+  let selectedAccounts: [BraveWallet.AccountInfo]
+  var showsSelectAllButton: Bool
+  var selectAccount: (BraveWallet.AccountInfo) -> Void
+  
+  init(
+    navigationTitle: String,
+    allAccounts: [BraveWallet.AccountInfo],
+    selectedAccounts: [BraveWallet.AccountInfo],
+    showsSelectAllButton: Bool,
+    selectAccount: @escaping (BraveWallet.AccountInfo) -> Void
+  ) {
+    self.navigationTitle = navigationTitle
+    self.allAccounts = allAccounts
+    self.selectedAccounts = selectedAccounts
+    self.showsSelectAllButton = showsSelectAllButton
+    self.selectAccount = selectAccount
+  }
+  
+  private var allSelected: Bool {
+    allAccounts.allSatisfy({ account in
+      selectedAccounts.contains(where: { $0.id == account.id && $0.coin == account.coin })
+    })
+  }
+  
+  private func selectAllButtonTitle(_ allSelected: Bool) -> String {
+    if allSelected {
+      return Strings.Wallet.deselectAllButtonTitle
+    }
+    return Strings.Wallet.selectAllButtonTitle
+  }
+  
+  var body: some View {
+    ScrollView {
+      LazyVStack(spacing: 0) {
+        HStack {
+          Text(Strings.Wallet.accountsPageTitle)
+            .font(.body.weight(.semibold))
+            .foregroundColor(Color(uiColor: WalletV2Design.textPrimary))
+          Spacer()
+          if showsSelectAllButton {
+            Button(action: {
+              if allSelected { // deselect all
+                allAccounts.forEach(selectAccount)
+              } else { // select all
+                let unselectedAccounts = allAccounts
+                  .filter { account in
+                    !selectedAccounts.contains(
+                      where: { $0.id == account.id && $0.coin == account.coin }
+                    )
+                  }
+                unselectedAccounts.forEach(selectAccount)
+              }
+            }) {
+              Text(selectAllButtonTitle(allSelected))
+                .font(.callout.weight(.semibold))
+                .foregroundColor(Color(uiColor: WalletV2Design.textInteractive))
+            }
+          }
+        }
+        .padding(.horizontal)
+        .padding(.vertical, 12)
+      }
+      ForEach(allAccounts) { account in
+        AccountListRowView(
+          account: account,
+          selectedAccounts: selectedAccounts
+        ) {
+          selectAccount(account)
+        }
+      }
+    }
+    .listBackgroundColor(Color(uiColor: WalletV2Design.containerBackground))
+    .navigationTitle(navigationTitle)
+    .navigationBarTitleDisplayMode(.inline)
+  }
+}
+
+private struct AccountListRowView: View {
+  
+  var account: BraveWallet.AccountInfo
+  var selectedAccounts: [BraveWallet.AccountInfo]
```

Comment:
i dont think `AccountListRowView` needs to know what `selectedAccounts` we can compute `isSelected` boolean outside of this struct and pass it via init

Reply:
Thanks, I've also updated `NetworkListRowView` to match in [89405ba](https://github.com/brave/brave-ios/pull/7665/commits/89405ba3ef0c8cb5b55ce9ec8f8f7c07643c9833) 👍.